src/js/reducers.js

```diff
@@ -456,27 +435,25 @@ function ui(state = {}, action) {
 
             if (action.nightColorScheme) {
                 app.nightColorScheme = action.nightColorScheme
-            }            
+            }
+            newState[action.appID] = app            
             return newState
-        case Actions.REGISTER_APPLICATION:
-            var newState = { ...state }         
+        case Actions.REGISTER_APPLICATION:        
             if (!newState[action.appID]) {
               newState[action.appID] = newAppState()
             }
-            var app = newState[action.appID]
-            if (app.displayLayout == null) {
+            app = newState[action.appID]
```

Comment:
i'm pretty sure this line and the three above it are unnecessary with new changes to code before the switch